File: 0050-powx-n/0050-powx-n.cpp
class Solution {
public:
    double solve(double x,long long n){
        double ans=1;
        while(n>0)
        {
            int c=n%2;
            if(c==1) ans=ans*x;
            x=x*x;
            n=n>>1;
        }
        return ans;
    }
    double myPow(double x, int n) {
         if(x == 1) return 1;
         double ans=solve(x,abs(n));
         
        if(n<0) return 1/ans;
         
        return ans;
    }
};

File: 0050-powx-n/0050-powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        def fun(x,n):
            if n==0: return 1
            num=fun(x,n//2)
            num*=num
            if n%2==0: return num
            return num*x
                
        num=fun(x,abs(n))
        if n<0:
            return 1/num
        return num